docker/docker_client.go

```diff
@@ -428,15 +379,34 @@ func getAuth(ctx *types.SystemContext, registry string) (string, string, error)
 }
 
 type pingResponse struct {
-	WWWAuthenticate string
-	APIVersion      string
-	scheme          string
+	APIVersion string
+	scheme     string
+}
+
+func (c *dockerClient) saveChallenges(chs []challenge, u *url.URL) {
+	urlCopy := url.URL{
+		Path:   u.Path,
+		Host:   u.Host,
+		Scheme: u.Scheme,
+	}
+	normalizeURL(&urlCopy)
+	c.challenges[urlCopy.String()] = chs
+}
+
+func (c *dockerClient) getChallenges(req *http.Request) []challenge {
+	ping := url.URL{
+		Host:   req.URL.Host,
+		Scheme: req.URL.Scheme,
+		Path:   "/v2/",
```

Comment:
funny thing is that #191 actually fixed https://bugzilla.redhat.com/show_bug.cgi?id=1413987 but then #195 broke it again :)